src/smtml/expr.ml

```diff
@@ -187,12 +187,13 @@ let get_symbols (hte : t list) =
 let negate_relop (hte : t) : (t, string) Result.t =
   let e =
     match view hte with
-    | Relop (ty, Eq, e1, e2) -> Ok (Relop (ty, Ne, e1, e2))
+    | Relop (ty, Eq, e1, e2) -> (* Ok (Relop (ty, Ne, e1, e2)) *)
+       Ok (Relop (ty, Ne, make (Binop (ty,Sub,e1,e2)), make (Val (Num (I32 0l)))))
```

Comment:
Are you sure that `I32 0l` is correct? What if `e1` and `e2` are of the type `I64` or anything else?

Assuming that `ty` in the pattern matching is the polymorphic equality's type parameter, `(ty,Sub,e1,e2)` is correct. But if that assumption is wrong, and `ty` is actually the type of the resulting expression, then the `ty` in the pattern matching is a boolean and the `ty` in  `(ty,Sub,e1,e2)` should be the type of `e1` (or `e2` which should be the same).

Reply:
Yeah, maybe you can use the function `ty` to get the type of the expressions?. (also, is this normalization safe for floats? Float.nan != 0. so it might be?)

Reply:
For floats, I am not sure, there are probably tricky corner cases, one i can think of: 
- `+inf = +inf` is true
- `+inf - +inf = 0` is false because `+inf - +inf` is NaN